src/overlays/actors/ovl_En_Honotrap/z_en_honotrap.c

```diff
@@ -24,59 +50,442 @@ const ActorInit En_Honotrap_InitVars = {
+        shieldVec.y = -player->shieldMf.zy;
+        shieldVec.z = -player->shieldMf.zz;
+        EnHonotrap_GetNormal(&shieldNorm, &shieldVec);
+
+        tempVel = this->actor.velocity;
+        Math3D_Vec3fReflect(&tempVel, &shieldNorm, &this->actor.velocity);
+        this->actor.speedXZ = this->speedMod * 0.5f;
+        this->actor.posRot.rot.y = Math_Atan2S(this->actor.velocity.z, this->actor.velocity.x);
+        EnHonotrap_SetupFlameVanish(this);
+    } else if (this->collider.tris.base.atFlags & 2) {
+        this->actor.velocity.y = this->actor.speedXZ = 0.0f;
+        EnHonotrap_SetupFlameVanish(this);
+    } else if (this->timer <= 0) {
+        EnHonotrap_SetupFlameVanish(this);
+    } else {
+        EnHonotrap_FlameCollisionCheck(this, globalCtx);
+        if (ready) {
+            EnHonotrap_SetupFlameChase(this);
+        }
+    }
+}
+
+void EnHonotrap_SetupFlameChase(EnHonotrap* this) {
+    this->actionFunc = EnHonotrap_FlameChase;
+
+    this->actor.velocity.x = this->actor.velocity.y = this->actor.velocity.z = this->actor.speedXZ = 0.0f;
+    this->actor.posRot.rot.x = this->actor.posRot.rot.y = this->actor.posRot.rot.z = 0;
+
+    this->timer = 100;
+}
+
+void EnHonotrap_FlameChase(EnHonotrap* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Player* player;
+    Vec3s shieldRot;
+
+    Math_ScaledStepToS(&this->actor.posRot.rot.y, this->actor.yawTowardsLink, 0x300);
+    Math_StepToF(&this->actor.speedXZ, 3.0f, 0.1f);
+    this->actor.gravity = (-this->actor.yDistToLink < 10.0f) ? 0.08f : -0.08f;
+    func_8002D868(&this->actor);
+    if (this->actor.velocity.y > 1.0f) {
+        this->actor.velocity.y = 1.0f;
+    }
+    this->actor.velocity.y *= 0.95f;
+    func_8002D7EC(&this->actor);
+    func_8002E4B4(globalCtx, &this->actor, 7.0f, 10.0f, 0.0f, 0x1D);
+    if (this->collider.cyl.base.atFlags & 4) {
+        player = PLAYER;
+        func_800D20CC(&player->shieldMf, &shieldRot, false);
+        this->actor.posRot.rot.y = ((shieldRot.y * 2) - this->actor.posRot.rot.y) + 0x8000;
+        EnHonotrap_SetupFlameVanish(this);
+    } else if (this->collider.cyl.base.atFlags & 2) {
+        this->actor.speedXZ *= 0.1f;
+        this->actor.velocity.y *= 0.1f;
+        EnHonotrap_SetupFlameVanish(this);
+    } else if ((this->actor.bgCheckFlags & 8) || (this->timer <= 0)) {
+        EnHonotrap_SetupFlameVanish(this);
+    } else {
+        EnHonotrap_FlameCollisionCheck(this, globalCtx);
+    }
+}
+
+void EnHonotrap_SetupFlameVanish(EnHonotrap* this) {
+    this->actionFunc = EnHonotrap_FlameVanish;
+}
+
+void EnHonotrap_FlameVanish(EnHonotrap* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 ready = Math_StepToF(&this->actor.scale.x, 0.0001f, 0.00015f);
+
+    this->actor.scale.z = this->actor.scale.y = this->actor.scale.x;
+    Actor_MoveForward(&this->actor);
+    func_8002E4B4(globalCtx, &this->actor, 7.0f, 10.0f, 0.0f, 0x1D);
+    if (ready) {
+        Actor_Kill(&this->actor);
+    }
```

Comment:
this seems more like finished to me than ready, tho i get what youre going for with "ready to vanish"

Reply:
I'm probably sticking with it for consistency.